### PR TITLE
Update Secrets container to use Reusable Table component

### PR DESCRIPTION
**Pull request by carlos-logro**
# Changes
Before tackling issue #816 I decided to update the Secrets container to use the Reusable Table, delete the SecretsTable component, and move all relevant test cases to Secrets.test. Lastly, since the Table component returns `getBatchActionProps().onCancel` when a batch action button is pressed, the secret delete action was updated to accept this method and use it when selected secrets are deleted successfully. 

This change also helps this issue progress a little: #708 

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/dashboard/blob/master/CONTRIBUTING.md)
for more details._

/cc @AlanGreene @a-roberts

### CODE REVIEW

File: src/containers/Secrets/Secrets.js [hunk 1]
```diff
@@ -154,11 +187,39 @@ export /* istanbul ignore next */ class Secrets extends Component {
...
+            {
+              id: 'dashboard.emptyState.selectedNamespace',
+              defaultMessage: 'No {kind} under namespace {selectedNamespace}'
+            },
+            { kind: 'Secrets', selectedNamespace }
+          )}
+          batchActionButtons={[
+            {
+              onClick: this.handleDeleteSecretClick,
+              text: 'Delete',
```
Comment (AlanGreene) [inline src/containers/Secrets/Secrets.js]: need to use `intl` for these user-facing strings

Comment (carlos-logro) [reply src/containers/Secrets/Secrets.js]: updated.

**Comment by carlos-logro** [comment]
@AlanGreene yeah I noticed this and doubt maybe the new css update will fix it but that wasn't the case... do you want me to include that css update here or in a seperate PR?

**Comment by AlanGreene** [comment]
In this PR would be best, otherwise we're introducing a known issue that should be easily avoided.

**Pull request merged**